### ISSUE TITLE
Run GHA on Windows, drop Appveyor test

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#5998]
Now we have 3 different apps for CI: GHA, Appveyor, and GLCI (this is down from 4 since we dropped Travis).

AIUI, we only need Appveyor to run CI on Windows, but GHA supports this. So we should add a Windows test to GHA and drop Appveyor for simplicity.

**Comment by jangorecki** [comment]
Windows is tested in GLCI so we could just drop appveyor.

**Comment by ben-schwen** [comment]
Well if GHA supports Windows I'm in favor of using it to build it in our CI. Should we build it on R-current or R-devel? Will file the PR later

**Comment by MichaelChirico** [comment]
> Windows is tested in GLCI so we could just drop appveyor.

hmm but we still need PRs CI to fail if they're not working on windows, otherwise I would've merged #5995 prematurely. is it possible for GLCI to post a GitHub comment if it fails?

**Comment by ben-schwen** [comment]
But AFAIU GLCI is only run once nightly on master?

**Comment by jangorecki** [comment]
yes GLCI run once, and does not fail as a GLCI, it builds the site and publish results of failed tests jobs. So the individual test jobs needs to be reported back to GH.

**Comment by ben-schwen** [comment]
> Yes GLCI run once a day, then it make sense to have some win job here then. GLCI does not fail as a GLCI, it builds the site and publish results of failed tests jobs. So the individual test jobs would have to report back to GH.

Yes and if we merge multiple PRs on a single day then the report will always end up in the latest merged PR or we would need to run a second time.
